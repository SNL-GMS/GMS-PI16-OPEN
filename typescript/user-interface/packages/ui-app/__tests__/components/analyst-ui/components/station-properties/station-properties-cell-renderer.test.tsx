import { shallow } from 'enzyme';
import * as React from 'react';

import {
  getColumnPosition,
  StationPropertiesCellRenderer,
  StationPropertiesTableCellRendererComponent,
  StationPropertiesTableCellRendererProps
} from '../../../../../src/ts/components/analyst-ui/components/station-properties/station-properties-cell-renderer';

window.ResizeObserver = jest.fn(() => {
  return { observe: jest.fn(), disconnect: jest.fn(), unobserve: jest.fn() };
});
const globalAny: any = global;
globalAny.ResizeObserver = window.ResizeObserver;
globalAny.DOMRect = jest.fn(() => ({}));

describe('station-properties-cell-renderer', () => {
  const rendererComponentProps: StationPropertiesTableCellRendererProps = {
    className: 'callsName',
    heightCSS: '100px',
    isNumeric: true,
    shouldCenterText: true,
    tooltipMsg: 'tooltip',
    value: 'some value',
    leftChild: <div />
  };
  const rendererProps: any = {
    valueFormatted: 'valueFormatted',
    value: 'value',
    colDef: {
      colId: 100
    },
    columnApi: {
      getAllDisplayedColumns: jest.fn(() => {
        return [{ getColId: jest.fn(() => 100) }];
      })
    }
  };
  const rendererPropsLast: any = {
    valueFormatted: 'valueFormatted',
    value: 'value',
    colDef: {
      colId: 100
    },
    columnApi: {
      getAllDisplayedColumns: jest.fn(() => {
        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
        return [{ getColId: jest.fn(() => 101) }, { getColId: jest.fn(() => 100) }];
      })
    }
  };
  test('is defined', () => {
    expect(StationPropertiesTableCellRendererComponent).toBeDefined();
    expect(StationPropertiesCellRenderer).toBeDefined();
    expect(getColumnPosition).toBeDefined();
  });
  test('can mount StationPropertiesTableCellRendererComponent', () => {
    const wrapper = shallow(
      // eslint-disable-next-line react/jsx-props-no-spreading
      <StationPropertiesTableCellRendererComponent {...rendererComponentProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('can mount StationPropertiesCellRenderer', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const wrapper = shallow(<StationPropertiesCellRenderer {...rendererProps} />);
    expect(wrapper).toMatchSnapshot();
  });
  test('getColumnPosition can determine first and last', () => {
    expect(getColumnPosition(rendererPropsLast)).toEqual('last');
    expect(getColumnPosition(rendererProps)).toEqual('first');
  });
});
