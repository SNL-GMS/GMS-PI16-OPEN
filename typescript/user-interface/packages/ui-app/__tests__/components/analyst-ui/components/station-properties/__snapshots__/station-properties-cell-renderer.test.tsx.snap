// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`station-properties-cell-renderer can mount StationPropertiesCellRenderer 1`] = `
<StationPropertiesTableCellRendererComponent
  data-col-position="first"
  isNumeric={false}
  value="valueFormatted"
/>
`;

exports[`station-properties-cell-renderer can mount StationPropertiesTableCellRendererComponent 1`] = `
<div
  className="station-properties-table-cell callsName"
  data-cy="station-properties-table-cell"
  style={
    Object {
      "height": "100px",
      "overflow": "hidden",
    }
  }
  title="tooltip"
>
  <div />
  <div
    className="station-properties-table-cell__value--numeric station-properties-table-cell__value--center station-properties-table-cell__value"
    data-cy="station-properties-table-cell__value"
  >
    <span>
      some value
    </span>
  </div>
</div>
`;
